### ISSUE TITLE
Hashing Structures helpers

### ISSUE THREAD

**Issue by gakonst** [scipr-lab/zexe#113]
Depends on #104 

It'd be nice to have a helper method as an extension trait over `CanonicalSerialize` which calls a cryptographically secure hash function (maybe make it take a [`Digest`](https://docs.rs/digest/0.8.1/digest/trait.Digest.html)?) on the serialized input, and returns the fixed size hash of the provided data.
